api/pkg/transformer/types/table/table_test.go

```diff
@@ -771,3 +770,299 @@ func TestTable_NewRaw(t *testing.T) {
 		})
 	}
 }
+
+func TestTable_SliceRow(t *testing.T) {
+	tests := []struct {
+		name       string
+		inputTable *Table
+		startIdx   int
+		endIdx     int
+		want       *Table
+		wantErr    bool
+		errMessage string
+	}{
+		{
+			name: "start < end, end < table row length",
+			inputTable: New(
+				series.New([]int{1, 2, 3, 4, 5}, series.Int, "A"),
+				series.New([]string{"a", "b", "c", "d", "e"}, series.String, "B"),
+			),
+			startIdx: 1,
+			endIdx:   3,
+			want: New(
+				series.New([]int{2, 3}, series.Int, "A"),
+				series.New([]string{"b", "c"}, series.String, "B"),
+			),
+		},
+		{
+			name: "start < end, end == table row length",
+			inputTable: New(
+				series.New([]int{1, 2, 3, 4, 5}, series.Int, "A"),
+				series.New([]string{"a", "b", "c", "d", "e"}, series.String, "B"),
+			),
+			startIdx: 1,
+			endIdx:   5,
+			want: New(
+				series.New([]int{2, 3, 4, 5}, series.Int, "A"),
+				series.New([]string{"b", "c", "d", "e"}, series.String, "B"),
+			),
+		},
+		{
+			name: "start < end, end > table row length",
+			inputTable: New(
+				series.New([]int{1, 2, 3, 4, 5}, series.Int, "A"),
+				series.New([]string{"a", "b", "c", "d", "e"}, series.String, "B"),
+			),
+			startIdx:   1,
+			endIdx:     6,
+			wantErr:    true,
+			errMessage: "failed slice col: A due to: slice index out of bounds",
+		},
+		{
+			name: "start > end",
+			inputTable: New(
+				series.New([]int{1, 2, 3, 4, 5}, series.Int, "A"),
+				series.New([]string{"a", "b", "c", "d", "e"}, series.String, "B"),
+			),
+			startIdx:   2,
+			endIdx:     0,
+			wantErr:    true,
+			errMessage: "failed slice col: A due to: slice index out of bounds",
+		},
+		{
+			name: "start < 0",
+			inputTable: New(
+				series.New([]int{1, 2, 3, 4, 5}, series.Int, "A"),
+				series.New([]string{"a", "b", "c", "d", "e"}, series.String, "B"),
+			),
+			startIdx:   -1,
```

Comment:
Can we allow negative index? It'll be useful for selecting series entry without having to know the series length. E.g.
```python
>>> a = [1,2,3,4,5,6,7,8,9,10]
# return last item
>>> a[-1]
10

# return 2nd last item
>>> a[-2]
9

# return last 2 items
>>> a[-2:]
[9, 10]
```

Reply:
can, I'll update this

Reply:
# return 2nd last item
>>> a[-2]
what is the end index in this case ya?
>>> a[-2:]
for this case the end index is nil

Reply:
I think you should mimic Python behavior